<modul>
    <titelDeutsch>
        Methods of Artificial Intelligence in Life Sciences
    </titelDeutsch>
    <titelEnglisch>
        Methods of Artificial Intelligence in Life Sciences
    </titelEnglisch>
    <modulkategorie>
        7
    </modulkategorie>
    <studiengang>
        Master-Studiengang Informatik
    </studiengang>
    <gesamtLeistungspunkte>
        10 LP
    </gesamtLeistungspunkte>
    <veranstaltungen>
        <veranstaltung>
            <titel>
                Methods of Artificial Intelligence in Life Sciences
            </titel>
            <leistungspunkte>
                10 LP
            </leistungspunkte>
            <veranstaltungsformen>
                <veranstaltungsform>
                    <form>
                        Vorlesung „Methods of Artificial Intelligence in Life Sciences“
                    </form>
                    <semesterWochenStunden>
                        4
                    </semesterWochenStunden>
                </veranstaltungsform>
                <veranstaltungsform>
                    <form>
                        Übung
                    </form>
                    <semesterWochenStunden>
                        8
                    </semesterWochenStunden>
                </veranstaltungsform>
            </veranstaltungsformen>
            <beschreibung>
                <inhalte>
                    Lectures and Practical Work: We start with an introduction to convolutional Neural Networks and show how to apply them in the search for overrepresented motifs in DNA. We introduce the necessary data preprocessing steps and illustrate how motif information can be extracted out of the learned weights. We then introduce deep generative models, in particular Variational Autoencoders and autoregessive Models in combination with important deep learning concepts, such as Attention. We introduce the problem of RNA folding and the basics of the biophysical and biochemical mechanisms involved. We use a variant of Monte Carlo Tree Search to efficiently sample the space of possible folding structures and introduce various scoring functions that serve as reward signal. We show that the problem of RNA/Protein folding is tightly related to reinforcement learning. We introduce the concept of self-supervised learning and how it can be applied to detect anomalies in time series data that have been recorded by wearables for high risk patients.
                </inhalte>
                <lernergebnisse>
                    Students know how to implement machine learning algorithms in Pytorch and to run it on HPC. They understand the concepts of Deep Learning and are familiar with common Neural Network architectures, such as Convolutional Neural Networks, Autoregressive Models, and Transformer Networks. They are familiar with the predominant sampling methods, such as Important Sampling, MCMC, and Monte Carlo Tree search. They understand the basics of the protein biosynthesis and the problem of predicting 3d RNA/Protein Structure from DNA sequence. They understand the concept of multiple sequence alignments, their relation to Evolutionary Biology, and how it can be used to increase prediction of 3d folding structure of Biomolecules.
                </lernergebnisse>
                <literatur>
                    TBD
                </literatur>
                <verwendbarkeit>
                    * Wahlpflichtbereich Praktische oder Technische Informatik
                    * Schwerpunktbereich
                    * Individuelle Ergänzung
                    * Wahlbereich im Master-Studiengang Artificial Intelligence and Data Science
                </verwendbarkeit>
                <voraussetzungenBestehen>
                    * Regular and active participation in the exercises
                    * Passing the examination
                    * Written examination
                </voraussetzungenBestehen>
                <haeufigkeit>
                    Each summer term
                </haeufigkeit>
                <sprache>

                </sprache>
            </beschreibung>
            <voraussetzungenTeilnahme>
                Bachelor-Studierende müssen folgende Module erfolgreich abgeschlossen haben:

                * „Programmierung”
                * „Rechnerarchitektur“
                * „Algorithmen und Datenstrukturen”
                * „Theoretische Informatik”
            </voraussetzungenTeilnahme>
            <semesters>
                <semester>

                </semester>
            </semesters>
            <zusatzfelder>
                <zusatzfeld>
                    <titel>

                    </titel>
                    <inhalt>

                    </inhalt>
                </zusatzfeld>
            </zusatzfelder>
        </veranstaltung>
    </veranstaltungen>
    <modulbeauftragte>
        Prof. Dr. Markus Kollmann
    </modulbeauftragte>
</modul>